### ISSUE TITLE
behavoir x-axis

### ISSUE THREAD

**Issue by christophbaur** [yvesmauron/bt#27]
x-axis ignores "x_order" when input column for "x" is not the same as for "x_order".  Creates some unwanted overlapping. It doesn't matter if "x" is unique or not.

**Comment by yvesmauron** [comment]
Hi

It seems that - if x is numeric - plotly assumes a continuous axis and orders it accordingly. Please check out the following example:

```
library(catmaply)

data("vbz")

df <- vbz[[2]]

# inverse ordering of the numerical x axis, nothing happens
df$trip_seq_order <- -as.numeric(df$trip_seq)

catmaply(
  df,
  x = trip_seq,
  x_order = trip_seq_order,
  y = stop_seq,
  y_order = stop_seq,
  z = occ_category
) 

# ordering with character x axis, correctly ordered
df$trip_seq <- as.character(df$trip_seq)
df$stop_seq <- as.character(df$stop_seq)

df$stop_seq_order <- as.numeric(df$stop_seq)
df$trip_seq_order <- as.numeric(df$trip_seq)

catmaply(
  df,
  x = trip_seq,
  x_order = trip_seq_order,
  y = stop_seq,
  y_order = stop_seq_order,
  z = occ_category
) 

# inverse ordering with character x axis, correctly ordered
df$trip_seq <- as.character(df$trip_seq)
df$stop_seq <- as.character(df$stop_seq)

df$stop_seq_order <- -as.numeric(df$stop_seq)
df$trip_seq_order <- -as.numeric(df$trip_seq)

catmaply(
  df,
  x = trip_seq,
  x_order = trip_seq_order,
  y = stop_seq,
  y_order = stop_seq_order,
  z = occ_category
) 

```

Does this apply for your issue as well; can you fix it by converting the x axis labels to character?

**Comment by christophbaur** [comment]
Well, it was may a bit to unspecific from my side. Let's have a look at these examples:

library(catmaply)
data("vbz")
df <- vbz[[2]]

#non unique 'label' for x (just two different types of vehicle) -> messed it up
catmaply(
  df,
  x = vehicle,
  x_order = trip_seq,
  y = stop_name,
  y_order = stop_seq,
  z = occ_category
) 

#unique 'label' -> will work
df <- df%>%
  mutate(really_fancy_label=paste0(trip_seq,"_",vehicle,"_",circulation_name))

catmaply(
  df,
  x = really_fancy_label,
  x_order = trip_seq,
  y = stop_name,
  y_order = stop_seq,
  z = occ_category
) 

#unique 'label' with slider=T -> 'full' x-axis visible, strange behavior
catmaply(
  df,
  x = really_fancy_label,
  x_order = trip_seq,
  y = stop_name,
  y_order = stop_seq,
  z = occ_category,
  slider=T
  )

**Comment by yvesmauron** [comment]
Just verified the issue; it seems to be an ordering issue the slider feature runs into when creating the traces. Potential fix would be to use the numerical ordering (`x_order`) of the x axis when building the plot and overwrite it in the layout using `tickmode`, `tickvals` and `ticktext`.

**Comment by yvesmauron** [comment]
Issue being fixed in PR #30 ; check there for progress.
